2007-11-05  Ulrich Drepper  <drepper@redhat.com>

	* time/tzfile.c (__tzfile_read): Fix check for version 0 data files.
	Patch by Szymon Siwek <sls@poczta.wp.pl>.

===================================================================
RCS file: /cvs/glibc/libc/time/tzfile.c,v
retrieving revision 1.69
retrieving revision 1.70
diff -u -r1.69 -r1.70
--- libc/time/tzfile.c	2007/10/17 14:44:26	1.69
+++ libc/time/tzfile.c	2007/11/06 01:03:43	1.70
@@ -381,7 +381,7 @@
       else
 	tzspec[tzspec_len - 1] = '\0';
     }
-  else if (sizeof (time_t) == 4 && tzhead.tzh_version != '\0')
+  else if (sizeof (time_t) == 4 && tzhead.tzh_version[0] != '\0')
     {
       /* Get the TZ string.  */
       if (__builtin_expect (fread_unlocked ((void *) &tzhead, sizeof (tzhead),
